<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>ENV222 oral presentation</title>
    <meta charset="utf-8" />
    <meta name="author" content="ID: 1931391" />
    <meta name="date" content="2023-03-08" />
    <script src="index_files/header-attrs/header-attrs.js"></script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# ENV222 oral presentation
]
.subtitle[
## <font size=6>Volcano plot in NGS(bioinformatics)</font>
]
.author[
### ID: 1931391
]
.date[
### 2023-03-08
]

---






### The detailed steps involved in *Next-Generation Sequencing*:

1. &lt;font color=Orange&gt;Library preparation&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;DNA or RNA samples are fragmented, size-selected, and ligated to adapters, which allows them to bind to the sequencing platform. This step is essential for generating high-quality reads.&lt;/span&gt;
2. &lt;font color=Orange&gt;Quality control&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The quality of the prepared library is assessed using various metrics such as concentration, purity, fragment size distribution, and quality scores. Libraries that pass the quality control are sequenced.&lt;/span&gt;
3. &lt;font color=Orange&gt;Sequencing&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The library is loaded onto a sequencer, which sequences millions of DNA or RNA fragments simultaneously. There are different sequencing technologies available, including Illumina, Ion Torrent, and PacBio.&lt;/span&gt;
4. &lt;font color=Orange&gt;Data processing&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;Raw sequencing data generated from the sequencer are processed to remove adapter sequences, low-quality reads, and contaminations. This step is essential to ensure the accuracy of downstream analyses.&lt;/span&gt;
5. &lt;font color=Orange&gt;Alignment&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The processed reads are aligned to a reference genome or transcriptome using specialized software such as Bowtie, BWA, STAR, or HISAT2. The alignment generates a BAM file, which stores the aligned reads' genomic coordinates.&lt;/span&gt;
6. &lt;font color=Orange&gt;Variant calling&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The aligned reads are analyzed to identify variants such as single-nucleotide polymorphisms (SNPs) and insertions/deletions (indels) using software such as GATK or FreeBayes.&lt;/span&gt;
7. &lt;font color=Orange&gt;Differential expression analysis&lt;/font&gt;&lt;font color=Red&gt;*&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The expression levels of genes are quantified using software such as HTSeq or FeatureCounts. The expression data are then analyzed to identify differentially expressed genes between different experimental conditions using software such as DESeq2 or edgeR.&lt;/span&gt;
8. &lt;font color=Orange&gt;Functional analysis&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The differentially expressed genes are annotated and analyzed to determine their biological functions, signaling pathways, and potential disease associations. This step is critical for understanding the biological relevance of the experimental results.&lt;/span&gt;
9. &lt;font color=Orange&gt;Visualization and interpretation&lt;/font&gt;: &lt;span style="font-size:18px;"&gt;The results from the functional analysis are visualized using tools such as heatmaps, &lt;font color=Red&gt;&lt;u&gt;volcano plots&lt;/u&gt;&lt;/font&gt;, and pathway diagrams. The results are then interpreted to draw conclusions and generate hypotheses.&lt;/span&gt;

---

## Paper from NCBI:

.left-column[
*&lt;font color=green&gt;Click the picture to open link of the NCBI paper:&lt;/font&gt;*
]

.right-column[
[![&lt;ENV222&gt; The wide visalization of science paper which was uesd volcano plot.](images/pnas-1620755114fig01.jpg)](https://pubmed.ncbi.nlm.nih.gov/28096348/)
]

**[Topic]** Transcriptomic, proteomic, and metabolomic landscape of positional memory &lt;br&gt;
            in the caudal fin of zebrafish

---

## High throughput data
Often, after a genetic biology experiment, we have a large number of data sets that we need to filter, and the most convenient way to do this is to use the P-value.

## Visualization using Volcano Plot
Using the volcano plot we can visualize the data with significant differences and have the flexibility to adjust the thresholds to suit the needs of the experiment. Here are some detailed explanations of the steps taken in the analysis of biological experimental data Source.

- Visualization of RNA-Seq results with Volcano Plot **[paper]**: [Visualization of RNA-Seq results with Volcano Plot](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html)

- Volcano plots with ggplot2 for differential gene expression **[video]**:
[Volcano plots with ggplot2 for differential gene expression](https://www.youtube.com/watch?v=sIRnaKo1aKE)

---

# Steps for differential gene detection

.left-column[
## First
## Second
## Third
]

.right-column[
- &lt;font color=Red&gt;[Preparing the inputs]&lt;/font&gt; Differential expressed results file (genes in rows, and 4 required columns: raw *P values*, *adjusted P values (FDR)*, *log fold change* and *gene labels*).
- &lt;font color=Red&gt;[Create volcano plot]&lt;/font&gt; Create a volcano plot coloring all significant genes. Typically, we will call genes significant here if FDR &lt; 0.01 and a log10 fold change of 3. These were the values used in the original paper for this dataset. Often, label the top 10 most significant genes with their gene names.
- &lt;font color=Red&gt;[Conclusion]&lt;/font&gt; The up-regulated differential genes and the down-regulated differential genes which was derived from the volcano plot then can be used for subsequent data analysis and comparison.
]

---

## Example(Find significant difference genes):

```r
# import the dataset
&gt; df &lt;- read.csv("volcano_plot_example_data.csv",
                 header = TRUE, stringsAsFactors = FALSE)
                 
# Preview the data structure                 
&gt; head(df)
  gene_name     logFC      P.Value    adj.P.Val
1   Csn1s2b -8.603611 3.830650e-15 6.053959e-11
2   Slc25a1 -4.124175 1.758595e-13 1.389642e-09
3    Atp2b2 -7.386986 4.836363e-13 2.432800e-09
4   Slc34a2 -4.177812 6.157428e-13 2.432800e-09
5     Acacb -4.314320 1.499977e-12 4.741129e-09
6       Egf -5.362664 2.116244e-12 5.574188e-09

# View the size of the dataset
&gt; dim(df)
15641     4
```

&lt;font color=Orange&gt;This CSV file is extremely large in data. It contains the four attributes gene name, log(fold change), P-value, adjusted P-value.&lt;/font&gt; 

---

## Data classification

```r
# Split the dataset and create a new column 'group'
&gt; df$group &lt;- ifelse(df$logFC&gt;=3&amp;df$P.Value&lt;=0.01, "Up", 
                     ifelse(df$logFC&lt;=-3&amp;df$P.Value&lt;=0.01, "Down", "Not sig"))

# Result
&gt; table(df$group)
   Down  Not sig    Up 
    228   15257     156 
```
&lt;br&gt;&lt;font color=Orange&gt;Transcripts were considered differentially expressed at *FDR &lt; 1%* and *fold change &gt; 3* between proximal and distal regions, whereas proteins were considered differentially expressed at *FDR &lt; 5%* and *fold change &gt; 2*.&lt;/font&gt; &lt;br&gt;&lt;br&gt;Here this dataset is a trans group so we use *log(fold change)&gt;3* and *P-value&lt;0.01* to split the dataset into three main parts, which are *Not significant*, *Up-regulated genes* and *Down-regulated genes*.

---

## Get the new dataset

```r
# Create the column 'pvalue_log10' then insert it into the table
&gt; df$pvalue_log10 &lt;- (-log10(df$P.Value))
&gt; df_sig &lt;- df[df$pvalue_log10&gt;=10,]
&gt; head(df_sig)
  gene_name     logFC      P.Value    adj.P.Val group pvalue_log10
1   Csn1s2b -8.603611 3.830650e-15 6.053959e-11  Down     14.41673
2   Slc25a1 -4.124175 1.758595e-13 1.389642e-09  Down     12.75483
3    Atp2b2 -7.386986 4.836363e-13 2.432800e-09  Down     12.31548
4   Slc34a2 -4.177812 6.157428e-13 2.432800e-09  Down     12.21060
5     Acacb -4.314320 1.499977e-12 4.741129e-09  Down     11.82392
6       Egf -5.362664 2.116244e-12 5.574188e-09  Down     11.67443

# Get the significant gene
&gt; dim(df_sig)
 21  6
```
&lt;font color=Orange&gt;Typically, we add log10() to the P-value so that we can get a more intuitive, linear data set that is easier to observe and analyse.&lt;/font&gt; &lt;br&gt;Here we use log10(P-value)&gt;=10 to the Significance criteria.

---

## Charting

```r
# Importing those packages we need
library(ggplot2)
library(ggrepel)

# Draw the volcano plot
ggplot(df, aes(x=logFC, y=-log10(P.Value))) + geom_point(aes(color=group)) +
  scale_color_manual(values=c("dodgerblue", "gray", "firebrick")) +
  geom_label_repel(data=df_sig,aes(x=logFC, y=-log10(P.Value), label=gene_name)) +
  labs(title = "Oral Presentation", subtitle = "&lt; ENV222 &gt;", 
       caption ="[Volcano plot R-demo for bioinformatics]", 
       y="-log10(p-value)", x="log(Fold Change)") +
  theme_bw()
```
&lt;br&gt;Set *logFC* as x_lab and *-log10(P.Value)* as y_lab. &lt;br&gt;Also, use different &lt;font color=Red&gt;colors&lt;/font&gt; to represent groups and add some &lt;font color=Red&gt;annotations&lt;/font&gt; for those significant genes. &lt;br&gt;By and large, some further diagrammatic &lt;font color=Red&gt;touches&lt;/font&gt; are necessary.

---

### Full code

```r
# dev.new()
getwd()

df &lt;- read.csv("volcano_plot_example_data.csv",
               header = TRUE, stringsAsFactors = FALSE)
head(df)
dim(df)
df$group &lt;- ifelse(df$logFC&gt;=3&amp;df$P.Value&lt;=0.01, "Up", 
                   ifelse(df$logFC&lt;=-3&amp;df$P.Value&lt;=0.01, "Down", "Not sig"))
table(df$group)
df$pvalue_log10 &lt;- (-log10(df$P.Value))
df_sig &lt;- df[df$pvalue_log10&gt;=10,]
head(df_sig)
dim(df_sig)
library(ggplot2)
library(ggrepel)
ggplot(df, aes(x=logFC, y=-log10(P.Value))) + geom_point(aes(color=group)) +
  scale_color_manual(values=c("dodgerblue", "gray", "firebrick")) +
  geom_label_repel(data=df_sig,aes(x=logFC, y=-log10(P.Value), label=gene_name)) +
  labs(title = "Oral Presentation", subtitle = "&lt; ENV222 &gt;", 
       caption ="[Volcano plot R-demo for bioinformatics]", 
       y="-log10(p-value)", x="log(Fold Change)") +
  theme_bw()
```

---

### Result

[![&lt;ENV222&gt; The wide visalization of volcano plot online.](images/R-volcano-plot.png)](https://i.ibb.co/C9sbPM7/Rplot.png)

---

class: inverse center middle

# [The end of the presentation]
## 😃 Thanks!
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%",
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
